core/payments/payments.go

```diff
@@ -192,6 +192,42 @@ func (s *service) sendEvent(status *pb.RpcMembershipGetStatusResponse) {
 	})
 }
 
+func getCacheExpireTime(dateEnds time.Time) time.Time {
+	const cacheLifetimeDur = 24 * time.Hour
+
+	// dateEnds can be 0
+	isExpired := time.Now().UTC().After(dateEnds)
+
+	timeNow := time.Now().UTC()
+	timeNext := timeNow.Add(cacheLifetimeDur)
+
+	// sub end < now OR no sub end provided (unlimited)
+	if isExpired {
+		log.Debug("setting cache to +1 day because subscription is isExpired")
+		return timeNext
+	}
+
+	// sub end >= now
+	// return min(sub end, now + 24h)
+	if dateEnds.Before(timeNext) {
+		log.Debug("setting cache to +1 day because subscription ends soon")
```

Comment:
It seems that this message is incorrect, as we are setting expiration time to less than one day. Moreover, cacheLifetimeDur can be changed, so we should consider different lifetime here